assets/src/components/notificationDrawer.tsx

```diff
@@ -4,44 +4,50 @@ import { StateDispatchContext } from "../contexts/stateDispatchContext"
 import useCurrentTime from "../hooks/useCurrentTime"
 import { markAllAsRead } from "../hooks/useNotificationsReducer"
 import { Notification } from "../realtime.d"
-import { closeNotificationDrawer, returnToPreviousView } from "../state"
+import {
+  closeView,
+  rememberNotificationDrawerScrollPosition,
+  returnToPreviousView,
+} from "../state"
 import { openVPPForNotification } from "./notifications"
 import { NotificationCard } from "./notificationCard"
 import ViewHeader from "./viewHeader"
 
 const NotificationDrawer = () => {
   const elementRef = useRef<HTMLDivElement | null>(null)
-  const { rememberScrollPosition, scrollPosition } =
-    useContext(NotificationsContext)
-  const [{ previousView }, dispatch] = useContext(StateDispatchContext)
+  const [{ previousView, notificationDrawerScrollPosition }, dispatch] =
+    useContext(StateDispatchContext)
 
   const [isInitialRender, setIsInitialRender] = useState<boolean>(true)
 
-  /* eslint-disable react-hooks/exhaustive-deps */
   useLayoutEffect(() => {
-    const restoreScrollPosition = isInitialRender
-    setIsInitialRender(false)
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    const currentElement = elementRef.current!
+    const element = elementRef.current
 
-    if (restoreScrollPosition && elementRef) {
-      const element = elementRef.current
-      if (element) {
-        element.scrollTop = scrollPosition
-      }
+    if (isInitialRender && element) {
+      setIsInitialRender(false)
+
+      element.scrollTop = notificationDrawerScrollPosition
     }
 
     return () => {
-      rememberScrollPosition(currentElement.scrollTop)
+      if (element) {
+        dispatch(rememberNotificationDrawerScrollPosition(element.scrollTop))
+      }
     }
-  }, [])
-  /* eslint-enable react-hooks/exhaustive-deps */
+  }, [isInitialRender, notificationDrawerScrollPosition, dispatch])
 
   return (
     <div className="m-notification-drawer" ref={elementRef}>
       <ViewHeader
         title="Notifications"
-        closeView={() => dispatch(closeNotificationDrawer())}
+        closeView={() => {
+          // reset scrollTop to avoid race condition with useEffect cleanup
```

Comment:
hmmmm, I see what you mean. An alternative that comes to mind would be instead of saving the scroll position as part of the cleanup, saving it as the user scrolls w/ an event listener (though maybe that would have a performance impact?) It could potentially allow for some more common code between the paths: something like a `useRememberScrollPosition` hook that takes the name of the dispatch action and does the event listener adding & cleanup under the hood. 

Though the approach you're taking here seems okay!

Reply:
That's a good point - I actually did try experimenting with a scroll event handler last week while I was working on fixing up the `useEffect` dependencies. I ran into something that cause me to abandon that approach but I don't remember exactly what. If I remember maybe I can give it another attempt with this logic issue in mind.